js/stanford_carousel.js

```diff
@@ -1,56 +1,76 @@
-jQuery(document).ready(function($) {
-  // Bootstrap Carousel
-  $('.carousel').attr('id', 'myCarousel');
-  $('.carousel .view-content').addClass('carousel-inner');
-  $('.carousel .item:nth-child(1)').addClass('active');
-  $('.carousel .item').attr('tabindex', '-1');
-  $('.carousel .item:nth-child(1)').attr('tabindex', '0');
-  // put the ol.carousel-indicators inside the div.carousel-dots
-  $('div.view-footer div.carousel-dots').append('<ol class="carousel-indicators"></ol>');
-  var dots = [];
-  // grab the contents of each h2 and pop it into the 'dots' array
-  $('.carousel-inner .views-row h2').each(function()
-  {
-    var header2 = $(this).text();
-    dots.push(header2);
-  }
-  );
-  // If we ended up with an empty array, that means there are no h2 elements.
-  // Therefore, it's a carousel without captions.
-  if(dots.length == 0) {
-    var i = 1;
-    $('.carousel-inner .views-row').each(function()
-    {
+(function($) {
+
+  $(document).ready(function($) {
+    // Bootstrap Carousel
+    var carousels = $('.carousel');
+    $.each(carousels, function(i, v) {
+      var myId = 'myCarousel-' + i;
+      var elem = $(v);
+      elem.attr('id', myId);
+      elem.find('.carousel-controls > button').attr('data-target', "#" + myId);
+      stanford_carousel_add_dots(elem, myId, i);
+    });
+
+    $('.carousel .view-content').addClass('carousel-inner');
+    $('.carousel .item:nth-child(1)').addClass('active');
+    $('.carousel .item').attr('tabindex', '-1');
+    $('.carousel .item:nth-child(1)').attr('tabindex', '0');
+
+  });
+
+
+
+  /**
+   * Adds nav dots to the carousel
+   * @param  {[type]} elem jquery selector object
+   * @param  {[type]} id css id name eg: myCar-0
+   * @param  {[type]} key the index id key
+   */
+  function stanford_carousel_add_dots($elem, $id, $key) {
+
+    // put the ol.carousel-indicators inside the div.carousel-dots
+    $elem.find('div.view-footer div.carousel-dots').append('<ol class="carousel-indicators"></ol>');
+    var dots = [];
+
+    // If we ended up with an empty array, that means there are no h2 elements.
+    // Therefore, it's a carousel without captions.
+    $elem.find('.views-row').each(function(i, v) {
       // Call it "Slide 1", "Slide 2" ,etc.
       var header = "Slide " + i;
```

Comment:
i is 0 in this case, leading the header to start with "Slide 0". This needs to be i+1.